apps/opik-backend/src/main/java/com/comet/opik/domain/stats/StatsMapper.java

```diff
@@ -28,6 +28,13 @@ public static ProjectStats mapProjectStats(Row row, String entityCountLabel) {
                 .add(new ProjectStats.CountValueStat("metadata", row.get("metadata", Long.class)))
                 .add(new ProjectStats.AvgValueStat("tags", row.get("tags", Double.class)));
 
+        BigDecimal totalEstimatedCost = row.get("total_estimated_cost_avg", BigDecimal.class);
+        if (totalEstimatedCost == null) {
+            totalEstimatedCost = BigDecimal.ZERO;
+        }
+
+        stats.add(new ProjectStats.AvgValueStat("total_estimated_cost", totalEstimatedCost.doubleValue()));
```

Comment:
At the end of the day, if we call `doubleValue` we end up losing precision, potentially. 

I think we mislead you with the previous comments and made you waste effort. Apologies about that. Probably for an AVG cost, it's ok to lose precision, as it's an AVG.

The ideal alternative by the book is that you create another flavour of `AvgValueStat` that contains a BigDecimal, instead of a Double.

That increases the effort and it's probably out of the scope of this task. But keep it in mind for the future.

Reply:
I agree; I was trying to avoid the contract here. But we can revisit it in the future